tests/validation/tests/v3_api/test_create_ha.py

```diff
@@ -0,0 +1,181 @@
+        create_ha_target_group(443, "target-group-443-" + test_suffix)
+    targetGroup80Arn = targetGroup80["TargetGroups"][0]["TargetGroupArn"]
+    targetGroup443Arn = targetGroup443["TargetGroups"][0]["TargetGroupArn"]
+
+    # Create listeners for the load balancer, to forward to the target groups
+    AmazonWebServices().create_ha_nlb_listener(loadBalancerARN=lbArn,
+                                               port=80,
+                                               targetGroupARN=targetGroup80Arn)
+    AmazonWebServices().create_ha_nlb_listener(loadBalancerARN=lbArn,
+                                               port=443,
+                                               targetGroupARN=targetGroup443Arn)
+
+    targets = []
+    aws_nodes = \
+        AmazonWebServices().create_multiple_nodes(
+            3, test_suffix, wait_for_ready=True)
+    assert len(aws_nodes) == 3
+
+    for aws_node in aws_nodes:
+        print(aws_node.public_ip_address)
+        targets.append(aws_node.provider_node_id)
+
+    # Register the nodes to the target groups
+    targets_list = [dict(Id=target_id, Port=80) for target_id in targets]
+    AmazonWebServices().register_targets(targets_list,
+                                         targetGroup80Arn)
+    targets_list = [dict(Id=target_id, Port=443) for target_id in targets]
+    AmazonWebServices().register_targets(targets_list,
+                                         targetGroup443Arn)
+    return aws_nodes
+
+
+def install_cert_manager():
+    helm_certmanager_cmd = \
+        export_cmd + " && " + \
+        "kubectl apply -f " + \
+        "https://raw.githubusercontent.com/jetstack/cert-manager/" + \
+        "release-0.12/deploy/manifests/00-crds.yaml && " + \
+        "kubectl create namespace cert-manager && " + \
+        "helm_v3 repo add jetstack https://charts.jetstack.io && " + \
+        "helm_v3 repo update && " + \
+        "helm_v3 install cert-manager jetstack/cert-manager " + \
+        "--namespace cert-manager --version v0.12.0"
+
+    print(helm_certmanager_cmd)
+    out = run_command_with_stderr(helm_certmanager_cmd)
+    print(out)
+    time.sleep(120)
+
+
+def add_repo_create_namespace():
+    helm_repo_cmd = \
+        export_cmd + " && helm_v3 repo add rancher-latest " + \
+        "https://releases.rancher.com/server-charts/latest && " + \
+        "helm_v3 repo update"
+
+    print(helm_repo_cmd)
+    out = run_command_with_stderr(helm_repo_cmd)
+    print(out)
+
+    helm_init_cmd = \
+        export_cmd + \
+        " && kubectl create namespace cattle-system"
+    print(helm_init_cmd)
+    out = run_command_with_stderr(helm_init_cmd)
+    print(out)
+
+
+def install_rancher_self_signed():
+    helm_rancher_cmd = \
+        export_cmd + " && helm_v3 install rancher rancher-latest/rancher " + \
+        "--version " + RANCHER_CHART_VERSION + " " \
+        "--namespace cattle-system " + \
+        "--set hostname=" + RANCHER_HA_HOSTNAME
```

Comment:
@bmdepesa Can we have the rancher repo also parameterized so we are able to deploy from rancher-stable/rancher-alpha/rancher-latest ?

Reply:
Added `RANCHER_HELM_REPO` -- user can supply one of `latest, stable, alpha`